src/tree.rs

```diff
@@ -436,6 +438,63 @@ impl<T> Tree<T> {
     }
 
     ///
+    /// Swaps two `Node`s including their children given their `NodeId`s.
+    ///
+    /// Returns an empty `Result` containing a `NodeIdError` if one occurred on either provided `Id`.
+    ///
+    /// Notes:
+    ///     - Both `NodeId`s are still valid after this process and are not swapped.
+    ///     - Keeps the positions of the `Node`s in their parents children collection.
+    ///
+    /// ``
```

Comment:
I dont think this makes sense here, and we already agreed on Option 2.

The reason it does not make more sense, is, that the swapped note should take their complete sub-tree with it, as long as that is possible to be as close to a real swap as we can.

You implementation idea sounds interesting, but I have already started another one. But I will keep that in mind.